### ISSUE TITLE
Hovering over a variable in the editor when it's too large causes the editor to crash

### ISSUE THREAD

**Issue by miklaskarjalainen** [godotengine/godot#64569]
### Godot version

4.0.alpha14

### System information

Arch Linux, AMD Radeon 6700 XT, Vulkan

## The Issue
The editor crashes when trying to get the stored data in a variable by hovering over it in the editor, on a crash or by using a breakpoint.

In the console i get the following error about 100 times before the crash:
![image](https://user-images.githubusercontent.com/47903526/185380204-2250f3b4-e93c-411f-add9-0656b3f13cc5.png)

This problem occurred in a project which used a lot of nested dictionaries.

### Steps to reproduce

https://user-images.githubusercontent.com/47903526/185382747-636531d6-cf4c-438b-a4f2-8aba2454657f.mp4

### Minimal reproduction project

[VariableHoverCrash.zip](https://github.com/godotengine/godot/files/9372936/VariableHoverCrash.zip)

**Comment by qarmin** [comment]
Cannot reproduce on Ubuntu 22.04, GTX 970 and Gnome 42
```
ERROR: GENERAL - Message Id Number: 4 | Message Id Name: NVIDIA
	Failed to create swapchain images (249561103)
	Objects - 1
		Object[0] - VK_OBJECT_TYPE_SWAPCHAIN_KHR, Handle 107339825976528
   at: _debug_messenger_callback (drivers/vulkan/vulkan_context.cpp:171)
ERROR: Condition "err" is true. Returning: ERR_CANT_CREATE
   at: _update_swap_chain (drivers/vulkan/vulkan_context.cpp:1753)
ERROR: Condition "err != OK" is true. Returning: ERR_CANT_CREATE
   at: _window_create (drivers/vulkan/vulkan_context.cpp:1461)
ERROR: Can't create a Vulkan window
   at: _create_window (platform/linuxbsd/display_server_x11.cpp:4554)
ERROR: Condition "!windows.has(p_window)" is true.
   at: window_set_flag (platform/linuxbsd/display_server_x11.cpp:2141)
ERROR: Condition "!windows.has(p_window)" is true.
   at: window_set_flag (platform/linuxbsd/display_server_x11.cpp:2141)
ERROR: Condition "!windows.has(p_window)" is true.
   at: window_set_flag (platform/linuxbsd/display_server_x11.cpp:2141)
ERROR: Condition "window_id == DisplayServer::INVALID_WINDOW_ID" is true.
   at: _make_window (scene/main/window.cpp:243)
```
looks that this may be related to graphic drivers/environment

**Comment by qarmin** [comment]
Well, after a while(1/2 minutes) without doing anything I got this crash which is probably unrelated to this issue
```
==62138==ERROR: AddressSanitizer: heap-use-after-free on address 0x60400095c8b0 at pc 0x00000b08defd bp 0x7f0b8c9d26f0 sp 0x7f0b8c9d26e0
READ of size 4 at 0x60400095c8b0 thread T10
    #0 0xb08defc in List<Array, DefaultAllocator>::size() const core/templates/list.h:467
    #1 0x1b4d0239 in RemoteDebuggerPeerTCP::_read_in() core/debugger/remote_debugger_peer.cpp:125
    #2 0x1b4d5656 in RemoteDebuggerPeerTCP::_poll() core/debugger/remote_debugger_peer.cpp:220
    #3 0x1b4d4c2b in RemoteDebuggerPeerTCP::_thread_func(void*) core/debugger/remote_debugger_peer.cpp:199
    #4 0x1ac1c934 in Thread::callback(Thread*, Thread::Settings const&, void (*)(void*), void*) core/os/thread.cpp:75
    #5 0x1ac222da in void std::__invoke_impl<void, void (*)(Thread*, Thread::Settings const&, void (*)(void*), void*), Thread*, Thread::Settings, void (*)(void*), void*>(std::__invoke_other, void (*&&)(Thread*, Thread::Settings const&, void (*)(void*), void*), Thread*&&, Thread::Settings&&, void (*&&)(void*), void*&&) /usr/include/c++/9/bits/invoke.h:60
    #6 0x1ac21c54 in std::__invoke_result<void (*)(Thread*, Thread::Settings const&, void (*)(void*), void*), Thread*, Thread::Settings, void (*)(void*), void*>::type std::__invoke<void (*)(Thread*, Thread::Settings const&, void (*)(void*), void*), Thread*, Thread::Settings, void (*)(void*), void*>(void (*&&)(Thread*, Thread::Settings const&, void (*)(void*), void*), Thread*&&, Thread::Settings&&, void (*&&)(void*), void*&&) /usr/include/c++/9/bits/invoke.h:95
    #7 0x1ac217b7 in void std::thread::_Invoker<std::tuple<void (*)(Thread*, Thread::Settings const&, void (*)(void*), void*), Thread*, Thread::Settings, void (*)(void*), void*> >::_M_invoke<0ul, 1ul, 2ul, 3ul, 4ul>(std::_Index_tuple<0ul, 1ul, 2ul, 3ul, 4ul>) /usr/include/c++/9/thread:244
    #8 0x1ac2144e in std::thread::_Invoker<std::tuple<void (*)(Thread*, Thread::Settings const&, void (*)(void*), void*), Thread*, Thread::Settings, void (*)(void*), void*> >::operator()() /usr/include/c++/9/thread:251
    #9 0x1ac21404 in std::thread::_State_impl<std::thread::_Invoker<std::tuple<void (*)(Thread*, Thread::Settings const&, void (*)(void*), void*), Thread*, Thread::Settings, void (*)(void*), void*> > >::_M_run() /usr/include/c++/9/thread:195
    #10 0x1db23493 in execute_native_thread_routine (/usr/bin/godot4s+0x1db23493)
    #11 0x7f0baab8eb42  (/lib/x86_64-linux-gnu/libc.so.6+0x94b42)
    #12 0x7f0baac209ff  (/lib/x86_64-linux-gnu/libc.so.6+0x1269ff)

0x60400095c8b0 is located 32 bytes inside of 40-byte region [0x60400095c890,0x60400095c8b8)
freed by thread T0 here:
    #0 0x7f0bab77446f in __interceptor_free ../../../../src/libsanitizer/asan/asan_malloc_linux.cc:122
    #1 0x1ac09742 in Memory::free_static(void*, bool) core/os/memory.cpp:168
    #2 0x258888c in DefaultAllocator::free(void*) core/os/memory.h:66
    #3 0xb0901c2 in void memdelete_allocator<List<Array, DefaultAllocator>::_Data, DefaultAllocator>(List<Array, DefaultAllocator>::_Data*) core/os/memory.h:124
    #4 0xb090073 in List<Array, DefaultAllocator>::erase(List<Array, DefaultAllocator>::Element const*) core/templates/list.h:432
    #5 0xb08edc7 in List<Array, DefaultAllocator>::pop_front() core/templates/list.h:352
    #6 0x1b4cb14a in RemoteDebuggerPeerTCP::get_message() core/debugger/remote_debugger_peer.cpp:49
    #7 0x1b4ae966 in RemoteDebugger::debug(bool, bool) core/debugger/remote_debugger.cpp:470
    #8 0x1b4d7f5f in ScriptDebugger::debug(ScriptLanguage*, bool, bool) core/debugger/script_debugger.cpp:99
    #9 0x57f6b0d in GDScriptLanguage::debug_break(String const&, bool) modules/gdscript/gdscript_editor.cpp:237
    #10 0x5b352c4 in GDScriptFunction::call(GDScriptInstance*, Variant const**, int, Callable::CallError&, GDScriptFunction::CallState*) modules/gdscript/gdscript_vm.cpp:3414
    #11 0x5507ecb in GDScriptInstance::callp(StringName const&, Variant const**, int, Callable::CallError&) modules/gdscript/gdscript.cpp:1586
    #12 0x11d7348c in bool Node::_gdvirtual__ready_call<false>() scene/main/node.h:241
    #13 0x11cdb5e9 in Node::_notification(int) scene/main/node.cpp:154
    #14 0x42db191 in Node::_notificationv(int, bool) scene/main/node.h:45
    #15 0x6cd6a4d in CanvasItem::_notificationv(int, bool) scene/main/canvas_item.h:45
    #16 0x118ff621 in Node2D::_notificationv(int, bool) scene/2d/node_2d.h:37
    #17 0x1c193ccd in Object::notification(int, bool) core/object/object.cpp:735
    #18 0x11cdcee5 in Node::_propagate_ready() scene/main/node.cpp:187
    #19 0x11cdc7c1 in Node::_propagate_ready() scene/main/node.cpp:179
    #20 0x11d36896 in Node::_set_tree(SceneTree*) scene/main/node.cpp:2576
    #21 0x11e11f7a in SceneTree::initialize() scene/main/scene_tree.cpp:400
    #22 0x259b604 in OS_LinuxBSD::run() platform/linuxbsd/os_linuxbsd.cpp:528
    #23 0x2582b8f in main platform/linuxbsd/godot_linuxbsd.cpp:72
    #24 0x7f0baab23d8f  (/lib/x86_64-linux-gnu/libc.so.6+0x29d8f)

previously allocated by thread T10 here:
    #0 0x7f0bab774868 in __interceptor_malloc ../../../../src/libsanitizer/asan/asan_malloc_linux.cc:144
    #1 0x1ac086e7 in Memory::alloc_static(unsigned long, bool) core/os/memory.cpp:75
    #2 0x2588869 in DefaultAllocator::alloc(unsigned long) core/os/memory.h:65
    #3 0x1ac0861b in operator new(unsigned long, void* (*)(unsigned long)) core/os/memory.cpp:44
    #4 0xb08e0eb in List<Array, DefaultAllocator>::push_back(Array const&) core/templates/list.h:284
    #5 0x1b4d215b in RemoteDebuggerPeerTCP::_read_in() core/debugger/remote_debugger_peer.cpp:148
    #6 0x1b4d5656 in RemoteDebuggerPeerTCP::_poll() core/debugger/remote_debugger_peer.cpp:220
    #7 0x1b4d4c2b in RemoteDebuggerPeerTCP::_thread_func(void*) core/debugger/remote_debugger_peer.cpp:199
    #8 0x1ac1c934 in Thread::callback(Thread*, Thread::Settings const&, void (*)(void*), void*) core/os/thread.cpp:75
    #9 0x1ac222da in void std::__invoke_impl<void, void (*)(Thread*, Thread::Settings const&, void (*)(void*), void*), Thread*, Thread::Settings, void (*)(void*), void*>(std::__invoke_other, void (*&&)(Thread*, Thread::Settings const&, void (*)(void*), void*), Thread*&&, Thread::Settings&&, void (*&&)(void*), void*&&) /usr/include/c++/9/bits/invoke.h:60
    #10 0x1ac21c54 in std::__invoke_result<void (*)(Thread*, Thread::Settings const&, void (*)(void*), void*), Thread*, Thread::Settings, void (*)(void*), void*>::type std::__invoke<void (*)(Thread*, Thread::Settings const&, void (*)(void*), void*), Thread*, Thread::Settings, void (*)(void*), void*>(void (*&&)(Thread*, Thread::Settings const&, void (*)(void*), void*), Thread*&&, Thread::Settings&&, void (*&&)(void*), void*&&) /usr/include/c++/9/bits/invoke.h:95
    #11 0x1ac217b7 in void std::thread::_Invoker<std::tuple<void (*)(Thread*, Thread::Settings const&, void (*)(void*), void*), Thread*, Thread::Settings, void (*)(void*), void*> >::_M_invoke<0ul, 1ul, 2ul, 3ul, 4ul>(std::_Index_tuple<0ul, 1ul, 2ul, 3ul, 4ul>) /usr/include/c++/9/thread:244
    #12 0x1ac2144e in std::thread::_Invoker<std::tuple<void (*)(Thread*, Thread::Settings const&, void (*)(void*), void*), Thread*, Thread::Settings, void (*)(void*), void*> >::operator()() /usr/include/c++/9/thread:251
    #13 0x1ac21404 in std::thread::_State_impl<std::thread::_Invoker<std::tuple<void (*)(Thread*, Thread::Settings const&, void (*)(void*), void*), Thread*, Thread::Settings, void (*)(void*), void*> > >::_M_run() /usr/include/c++/9/thread:195
    #14 0x1db23493 in execute_native_thread_routine (/usr/bin/godot4s+0x1db23493)

Thread T10 created by T0 here:
    #0 0x7f0bab6a1875 in __interceptor_pthread_create ../../../../src/libsanitizer/asan/asan_interceptors.cc:208
    #1 0x1db23758 in std::thread::_M_start_thread(std::unique_ptr<std::thread::_State, std::default_delete<std::thread::_State> >, void (*)()) (/usr/bin/godot4s+0x1db23758)
    #2 0x1ac1ce2a in Thread::start(void (*)(void*), void*, Thread::Settings const&) core/os/thread.cpp:91
    #3 0x1b4d42d5 in RemoteDebuggerPeerTCP::connect_to_host(String const&, unsigned short) core/debugger/remote_debugger_peer.cpp:188
    #4 0x1b4d5ea7 in RemoteDebuggerPeerTCP::create(String const&) core/debugger/remote_debugger_peer.cpp:238
    #5 0x1b46611d in EngineDebugger::initialize(String const&, bool, Vector<String>, void (*)()) core/debugger/engine_debugger.cpp:153
    #6 0x271d04a in Main::setup(char const*, int, char**, bool) main/main.cpp:1305
    #7 0x2582a7d in main platform/linuxbsd/godot_linuxbsd.cpp:61
    #8 0x7f0baab23d8f  (/lib/x86_64-linux-gnu/libc.so.6+0x29d8f)

SUMMARY: AddressSanitizer: heap-use-after-free core/templates/list.h:467 in List<Array, DefaultAllocator>::size() const
```

**Comment by Schweini07** [comment]
Can confirm the bug on Manjaro Linux with Intel HD Graphics 510:
```
Godot Engine v4.0.alpha.custom_build.0c5f25495 - https://godotengine.org
Vulkan API 1.2.0 - Using Vulkan Device #0: Intel - Intel(R) HD Graphics 510 (SKL GT1)
 
X Error of failed request:  BadDrawable (invalid Pixmap or Window parameter)
  Major opcode of failed request:  149 ()
  Minor opcode of failed request:  4
  Resource id in failed request:  0x6600075
  Serial number of failed request:  6026
  Current serial number in output stream:  6032
```
Godot reports a BadDrawable, and then continues to spit out a lot of errors:
```
ERROR: BUG: Unreferenced static string to 0: _update_scrollbars
   at: unref (core/string/string_name.cpp:129)
ERROR: BUG: Unreferenced static string to 0: hseparator
   at: unref (core/string/string_name.cpp:129)
ERROR: Method argument to Callable constructor must be a non-empty string
   at: Callable (core/variant/callable.cpp:278)
ERROR: BUG: Unreferenced static string to 0: _physics_process_internal
   at: unref (core/string/string_name.cpp:129)
ERROR: Method argument to Callable constructor must be a non-empty string
   at: Callable (core/variant/callable.cpp:278)
ERROR: BUG: Unreferenced static string to 0: _update_script_names
   at: unref (core/string/string_name.cpp:129)
ERROR: Method argument to Callable constructor must be a non-empty string
   at: Callable (core/variant/callable.cpp:278)
ERROR: BUG: Unreferenced static string to 0: PrevScene
   at: unref (core/string/string_name.cpp:129)
ERROR: Method argument to Callable constructor must be a non-empty string
   at: Callable (core/variant/callable.cpp:278)
ERROR: BUG: Unreferenced static string to 0: GuiToggleOnDisabledMirrored
   at: unref (core/string/string_name.cpp:129)
ERROR: Method argument to Callable constructor must be a non-empty string
   at: Callable (core/variant/callable.cpp:278)
ERROR: BUG: Unreferenced static string to 0: ShaderCompilation
   at: unref (core/string/string_name.cpp:129)
ERROR: Method argument to Callable constructor must be a non-empty string
   at: Callable (core/variant/callable.cpp:278)
ERROR: BUG: Unreferenced static string to 0: main_screen_changed
   at: unref (core/string/string_name.cpp:131)
ERROR: Method argument to Callable constructor must be a non-empty string
   at: Callable (core/variant/callable.cpp:278)
ERROR: BUG: Unreferenced static string to 0: scene_changed
   at: unref (core/string/string_name.cpp:131)
ERROR: Method argument to Callable constructor must be a non-empty string
   at: Callable (core/variant/callable.cpp:278)
ERROR: BUG: Unreferenced static string to 0: picker_created
   at: unref (core/string/string_name.cpp:131)
ERROR: Method argument to Callable constructor must be a non-empty string
   at: Callable (core/variant/callable.cpp:278)
ERROR: BUG: Unreferenced static string to 0: process_frame
   at: unref (core/string/string_name.cpp:131)
ERROR: Method argument to Callable constructor must be a non-empty string
   at: Callable (core/variant/callable.cpp:278)
ERROR: BUG: Unreferenced static string to 0: physics_frame
   at: unref (core/string/string_name.cpp:131)
ERROR: Method argument to Callable constructor must be a non-empty string
   at: Callable (core/variant/callable.cpp:278)
ERROR: BUG: Unreferenced static string to 0: interface_added
   at: unref (core/string/string_name.cpp:131)
ERROR: Method argument to Callable constructor must be a non-empty string
   at: Callable (core/variant/callable.cpp:278)
ERROR: BUG: Unreferenced static string to 0: button_up
   at: unref (core/string/string_name.cpp:131)
ERROR: Method argument to Callable constructor must be a non-empty string
   at: Callable (core/variant/callable.cpp:278)
ERROR: BUG: Unreferenced static string to 0: frame_pre_draw
   at: unref (core/string/string_name.cpp:131)
ERROR: Method argument to Callable constructor must be a non-empty string
   at: Callable (core/variant/callable.cpp:278)
ERROR: BUG: Unreferenced static string to 0: frame_post_draw
   at: unref (core/string/string_name.cpp:131)
ERROR: Method argument to Callable constructor must be a non-empty string
   at: Callable (core/variant/callable.cpp:278)
ERROR: BUG: Unreferenced static string to 0: editor_script_changed
   at: unref (core/string/string_name.cpp:131)
ERROR: Method argument to Callable constructor must be a non-empty string
   at: Callable (core/variant/callable.cpp:278)
ERROR: BUG: Unreferenced static string to 0: started
   at: unref (core/string/string_name.cpp:131)
ERROR: Method argument to Callable constructor must be a non-empty string
   at: Callable (core/variant/callable.cpp:278)
ERROR: BUG: Unreferenced static string to 0: version_changed
   at: unref (core/string/string_name.cpp:131)
ERROR: Method argument to Callable constructor must be a non-empty string
   at: Callable (core/variant/callable.cpp:278)
ERROR: Pages in use exist at exit in PagedAllocator: N7Variant5Pools11BucketLargeE
   at: ~PagedAllocator (./core/templates/paged_allocator.h:140)
ERROR: Method argument to Callable constructor must be a non-empty string
   at: Callable (core/variant/callable.cpp:278)
ERROR: Pages in use exist at exit in PagedAllocator: N7Variant5Pools12BucketMediumE
   at: ~PagedAllocator (./core/templates/paged_allocator.h:140)
ERROR: Method argument to Callable constructor must be a non-empty string
   at: Callable (core/variant/callable.cpp:278)
ERROR: Pages in use exist at exit in PagedAllocator: N7Variant5Pools11BucketSmallE
   at: ~PagedAllocator (./core/templates/paged_allocator.h:140)
ERROR: Method argument to Callable constructor must be a non-empty string
   at: Callable (core/variant/callable.cpp:278)
ERROR: BUG: Unreferenced static string to 0: settings_changed
   at: unref (core/string/string_name.cpp:131)
```
After that, Godot prints out this error about a 100 times maybe even more:
```
ERROR: Condition "!EditorSettings::get_singleton() || !EditorSettings::get_singleton()->has_setting(p_setting)" is true. Returning: Variant()
   at: _EDITOR_GET (editor/editor_settings.cpp:1073)
```

I feel that maybe a size limit for drawable characters during the hover could be the issue. But what's interesting is, that for @giffi-dev the limit was 3392 characters for the string and more than that would bring Godot to crash. But for me it also crashed with less characters than that, although I couldn't determine at which point the limit was. And for @qarmin the limit seems to be higher than 3392, maybe you could try it with more characters and see if Godot crashes for you then?

**Comment by reduz** [comment]
Appears fixed by #64710
Here screenshot:
![image](https://user-images.githubusercontent.com/6265307/212176684-961fdcb3-8b4d-4e97-baa7-a102866e5a9c.png)
Closing.

**Comment by Tech074** [comment]
In debug mode, I am getting an editor crash while mouse hovering on a String variable with high character length values.
The pop up preview of the value stored in the String variable will display, but then the editor crashes.
Godot 4.1 Stable 
![Capture](https://github.com/user-attachments/assets/26cfabf6-a7c0-4df4-b91e-6e852e1cac21)
